src/ui/components/controls/CreateButton.vue

```diff
@@ -71,27 +68,30 @@
             toggleCreateMenu: function () {
                 this.showCreateMenu = !this.showCreateMenu;
             },
+            showItemDescription: function (menuItem) {
+                this.selectedMenuItem = menuItem;
+            }
         },
         data: function() {
+            let createMenuItems = [];
+
+            this.openmct.types.listKeys().forEach(key => {
+                let menuItem = openmct.types.get(key).definition;
+
+                if (menuItem.creatable) {
+                    let menuItemTemplate = {
+                        name: menuItem.name,
+                        class: menuItem.cssClass,
+                        title: menuItem.description
+                    };
+
+                    createMenuItems.push(menuItemTemplate);
+                }
+            });
+
             return {
-                createMenuItems: [
-                    { name: 'Folder', class: 'icon-folder', title: 'Details will go here' },
-                    { name: 'Display Layout', class: 'icon-layout', title: 'Details will go here' },
-                    { name: 'Fixed Position Display', class: 'icon-box-with-dashed-lines', title: 'Details will go here' },
-                    { name: 'Overlay Plot', class: 'icon-plot-overlay', title: 'Details will go here' },
-                    { name: 'Stacked Plot', class: 'icon-plot-stacked', title: 'Details will go here' },
-                    { name: 'Telemetry Table', class: 'icon-tabular-realtime', title: 'Details will go here' },
-                    { name: 'Clock', class: 'icon-clock', title: 'Details will go here' },
-                    { name: 'Timer', class: 'icon-timer', title: 'Details will go here' },
-                    { name: 'Web Page', class: 'icon-page', title: 'Details will go here' },
-                    { name: 'Event Message Generator', class: 'icon-folder-new', title: 'Details will go here' },
-                    { name: 'Hyperlink', class: 'icon-chain-links', title: 'Details will go here' },
-                    { name: 'Notebook', class: 'icon-notebook', title: 'Details will go here' },
-                    { name: 'State Generator', class: 'icon-telemetry', title: 'Details will go here' },
-                    { name: 'Sine Wave Generator', class: 'icon-telemetry', title: 'Details will go here' },
-                    { name: 'Example Imagery', class: 'icon-image', title: 'Details will go here' },
-                    { name: 'Summary Widget', class: 'icon-summary-widget', title: 'Details will go here' }
-                ]
+                createMenuItems: createMenuItems,
```

Comment:
Data names are repetitive given that this is the create menu component.  Just name them "items" and "selected".